### PR TITLE
INTEG-491 : init mocks correctly in test class WikiSpaceActivityPublisherTest to avoid random failure

### PR DESCRIPTION
**Pull request by thomasdelhomenie**
The mocks of the test class WikiSpaceActivityPublisherTest were not init correctly. The PortalContainer method to get a component was actually mocked to return dependencies whereas this method is a static method, so it did not work correctly.
This made the test failed randomly depending on the state of the container.

This fix:
* inject dependencies of the tested class WikiSpaceActivityPublisher through the constructor to make the class more easily testable
* use annotations to init mocks
* split a test method which contained 2 tests in 2 methods

### CODE REVIEW

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 1]
```diff
@@ -344,16 +352,16 @@ private String getValueFromResourceBundle(String key, String[] params) {
...
     boolean isPublic = false;
-    if(permissions != null) {
-      for(PermissionEntry permissionEntry : permissions) {
-        if(permissionEntry.getId().equals(IdentityConstants.ANY.toString())) {
-          for(Permission permission : permissionEntry.getPermissions()) {
-            if(PermissionType.VIEWPAGE.toString().equals(permission.getPermissionType())) {
+    if (permissions != null) {
+      for (PermissionEntry permissionEntry : permissions) {
+        if (permissionEntry.getId().equals(IdentityConstants.ANY.toString())) {
+          for (Permission permission : permissionEntry.getPermissions()) {
```
Comment (seifeddineDaoud) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: "ANY" is already a string, there's no need to call "toString()" on it

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 2]
```diff
@@ -344,16 +352,16 @@ private String getValueFromResourceBundle(String key, String[] params) {
...
+            if (PermissionType.VIEWPAGE.toString().equals(permission.getPermissionType())) {
               isPublic = true;
```
Comment (seifeddineDaoud) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: comparisons between unrelated types

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 3]
```diff
@@ -117,18 +131,17 @@ private ExoSocialActivity generateActivity(Identity ownerStream,
...
     List<PageVersion> versions = wikiService.getVersionsOfPage(page);
-    if(versions != null && !versions.isEmpty()) {
+    if (versions != null && !versions.isEmpty()) {
       versionsTotal = versions.size();
     }
     templateParams.put(WIKI_PAGE_VERSION, String.valueOf(versionsTotal));
-    
+
     // Create page excerpt
     StringBuffer excerpt = new StringBuffer();
```
Comment (seifeddineDaoud) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: IMO, we replace  StringBuffer by StringBuilder

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 4]
```diff
@@ -183,12 +196,11 @@ private ExoSocialActivity generateActivity(Identity ownerStream,
...
                                "WikiUIActivity.msg.update-page-title",
-                               new String[]{ page.getTitle() },
+                               new String[] { page.getTitle() },
                                comment,
                                null);
         }
       } else if (PageUpdateType.MOVE_PAGE.equals(activityType)) {
-        WikiService wikiService = PortalContainer.getInstance().getComponentInstanceOfType(WikiService.class);
         List<BreadcrumbData> breadcrumbDatas = wikiService.getBreadcumb(wikiType, wikiOwner, pageId);
         StringBuffer breadcrumText = new StringBuffer();
```
Comment (seifeddineDaoud) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: IMO, we replace StringBuffer by StringBuilder

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 5]
```diff
@@ -203,24 +215,22 @@ private ExoSocialActivity generateActivity(Identity ownerStream,
...
   }
-  
+
   private String validateExcerpt(String excerpt) {
-    List<String> lines = Stream.of(excerpt.split("\n"))
-            .filter(line -> !line.trim().isEmpty()).collect(Collectors.toList());
+    List<String> lines = Stream.of(excerpt.split("\n")).filter(line -> !line.trim().isEmpty()).collect(Collectors.toList());
 
     Stream<String> sLines = lines.stream();
     StringBuffer result = new StringBuffer();
```
Comment (seifeddineDaoud) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: IMO, we replace StringBuffer by StringBuilder

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java (hunk 4 above)
Comment (thomasdelhomenie) [reply integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Done

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java (hunk 5 above)
Comment (thomasdelhomenie) [reply integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Done

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java (hunk 1 above)
Comment (thomasdelhomenie) [reply integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Done

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java (hunk 2 above)
Comment (thomasdelhomenie) [reply integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Done

**Pull request merged**